### DISCUSSION TITLE
Do we need to put Reword.reword() in fragments?

### DISCUSSION THREAD

**Discussion by adanecito** [B3nedikt/restring#112] [General]
I have not found an example using fragments. If I can get the getDelegate to work in my activity class do I need to do anything with any fragments later? Most examples are very simple but I have a more complicated use case.

Thanks!

**Comment by B3nedikt** [comment]
The example app uses fragments actually. You only need to do the getDelegate() in your activity class, and it will work in all fragments contained in this activity, you don't need to add anything to your fragments. Reword.reword(...) just tries to update the texts of whatever view you pass to it, if you want to update the whole view hierarchy just pass in the root view:

```kotlin
// The layout containing the views you want to localize
val rootView = window.decorView.findViewById<ContentFrameLayout>(android.R.id.content)
Reword.reword(rootView)
```
